### DISCUSSION TITLE
VS 2017 support after `windows-2016` is deprecated

### DISCUSSION THREAD

**Discussion by seladb** [actions/virtual-environments#5161] [Installed software]
Hi GitHub Actions team!

As you [announced](https://github.blog/changelog/2021-10-19-github-actions-the-windows-2016-runner-image-will-be-removed-from-github-hosted-runners-on-march-15-2022/) a few months ago `windows-2016` will be removed on Mar 15, 2022. 
I'm using this image to [build my project using VS 2017](https://github.com/seladb/PcapPlusPlus/blob/master/.github/workflows/build_and_test.yml#L338-L346) which as far as I can tell isn't supported in any of the newer Windows images.
What options do I have to continue building with VS 2017?

Thank you,
PcapPlusPlus maintainer

**Comment by miketimofeev** [comment]
Hi @seladb! The easiest way is to set up your own self-hosted runner with VS2017 on board.

**Reply by seladb** [reply]
But self hosted runners are not free which is an issue for open source projects who don't have funding...

**Reply by miketimofeev** [reply]
Another option is to install vs2017 build tools in runtime, but it will take some time, I'm afraid

**Reply by seladb** [reply]
Maybe it's possible to add VS 2017 build tools to the Windows image by default?

**Reply by miketimofeev** [reply]
@seladb Unfortunately we can't do it from the perspective of maintenance concerns

**Reply by seladb** [reply]
Just out of curiosity: can you elaborate on these concerns?

**Reply by miketimofeev** [reply]
Installation time and size for example. We try to keep images as small as possible. There are also some security concerns that I'm unable to share, sorry

**Reply by seladb** [reply]
Got it, thanks for the info!
